### PR TITLE
feat(core): apply all quota guard

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement every quota limit guard. (hard limit)

Sorry for the large PR, guards are in many different routes.

## Testing

Local tested

## Checklist

- [x] This PR is not applicable for the checklist

Just like other middlewares, it is not easy to test in every routes. So the unit tests for routers won't cover this.

There are unit tests for `quotaLibrary`.

### CODE REVIEW

File: packages/core/src/libraries/quota.ts [hunk 1]
```diff
@@ -6,35 +8,70 @@ import { getTenantSubscriptionPlan } from '#src/utils/subscription/index.js';
...
+  const { getLogtoConnectors } = connectorLibraray;
+
+  const tenantUsageQueries: Record<
+    keyof FeatureQuota,
+    (queryKey?: string) => Promise<{ count: number }>
+  > = {
+    applicationsLimit: countNonM2MApplications,
+    hooksLimit: getTotalNumberOfHooks,
+    machineToMachineLimit: countM2MApplications,
+    resourcesLimit: findTotalNumberOfResources,
```
Comment (xiaoyijun) [inline packages/core/src/libraries/quota.ts:42]: We should exclude the default management API.

File: packages/core/src/libraries/quota.ts [hunk 2]
```diff
@@ -55,6 +92,10 @@ export const createQuotaLibrary = (queries: Queries, cloudConnection: CloudConne
     const plan = await getTenantSubscriptionPlan(cloudConnection);
```
Comment (simeng-li) [inline packages/core/src/libraries/quota.ts:92]: nit: feels like setTenantSubscriptionPlan can extract as a middleware, and provides tenant subscription status in the context.  Could do later.

Comment (simeng-li) [inline packages/core/src/libraries/quota.ts:92]: nit: feels like setTenantSubscriptionPlan can extract as a middleware, and provides tenant subscription status in the context.  Could do later.

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: packages/core/src/libraries/quota.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/libraries/quota.ts:92]: Cool, LOG-6589

File: packages/core/src/libraries/quota.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/libraries/quota.ts]: rebased

**Pull request merged**